tests/test_csc.py

```diff
@@ -329,213 +329,136 @@ async def test_camera_cable_wrap_following(self):
+
+                assert self.mock_controller.command_queue.empty()
 
-            # Enable tracking and check mock axis controllers
-            await self.remote.cmd_startTracking.start(timeout=STD_TIMEOUT)
-            self.assertTrue(mock_azimuth.tracking_enabled)
-            self.assertTrue(mock_elevation.tracking_enabled)
-
-            # Slew and track until both axes are in position.
-            t0 = time.monotonic()
-            estimated_slew_time = 2  # seconds
-            tracking_task = asyncio.create_task(
-                self.track_target_loop(
-                    azimuth=initial_azimuth + 3,
-                    elevation=initial_elevation + 1,
-                    azimuth_velocity=0.003,
-                    elevation_velocity=0.003,
+                # Test regular tracking mode. CCW and MTRotator start in sync.
+                ccw_limits = mtmount.LimitsDict[mtmount.DeviceId.CAMERA_CABLE_WRAP]
+                # seconds until the CCW demand hits the position limit;
+                # make it large enough that we get a few target events
+                # before we reach the position limit.
+                trunc_time = 0.5
+                if do_max_limit:
+                    position_limit = ccw_limits.max_position
+                    velocity_limit = ccw_limits.max_velocity
+                else:
+                    position_limit = ccw_limits.min_position
+                    velocity_limit = -ccw_limits.max_velocity
+                velocity = velocity_limit * 1.1
+                position0 = position_limit - (velocity * trunc_time)
+                print(
+                    f"Position limit={position_limit}, velocity limit={velocity_limit}"
                 )
-            )
-            await self.assert_next_sample(
-                topic=self.remote.evt_elevationInPosition,
-                inPosition=True,
-                timeout=estimated_slew_time + STD_TIMEOUT,
-            )
-            await self.assert_next_sample(
-                self.remote.evt_azimuthInPosition,
-                inPosition=True,
-                timeout=estimated_slew_time + STD_TIMEOUT,
-            )
-            dt = time.monotonic() - t0
-            tracking_task.cancel()
-            print(f"Time to finish slew={dt:0.2f} seconds")
 
-            # Disable tracking and check axis controllers
-            await self.remote.cmd_stopTracking.start(timeout=STD_TIMEOUT)
-            self.assertFalse(mock_azimuth.tracking_enabled)
-            self.assertFalse(mock_elevation.tracking_enabled)
+                tai0 = utils.current_tai()
+                previous_tai = 0
+                while True:
+                    tai = utils.current_tai()
+                    # Work around non-monotonic clocks, which are
+                    # sometimes seen when running Docker on macOS.
+                    if tai < previous_tai:
+                        tai = previous_tai + 0.001
```

Comment:
Is this really enough to ensure that `tai` is larger than `previous_tai`? Wouldn't establishing the difference and adding that plus a small margin to tai be a better approach?

Reply:
Yes. The problem is that the system clock on Docker macOS has occasional small backward jumps. If a tracking command arrives with TAI <= previous commanded TAI the simulated tracking actuator rejects the command, which breaks the test. All that is required for these tests to pass reliably is that the commanded times be monotonically increasing.

Reply:
Yes, I understood that. I was merely wondering if adding `0.001` always is enough and if that could possibly change in the future. I guess that if it does we will deal with it then.